Add free-text option to "Other" license selection

josephsnyder opened the pull request:
Add a text box which will allow the submitter to add the name
of a different license when the "Other" licensing option is
selected during submission.

josephsnyder committed:
Add free-text option to "Other" license selection

Add a text box which will allow the submitter to add the name
of a different license when the "Other" licensing option is
selected during submission.

BetsyMcPhail commented:
If you select the license checkbox while 'Apache 2' is selected and then select 'Other', the 'Publish' button remains disabled.

Otherwise, works as expected.

josephsnyder committed:
Consolidate "Submit" checks

Consolidate the scattered checks to see if the "Submit" button
should be disabled into a single function.

Fix the submit check such that the button is disabled if the
text box for the free text value is empty.
